Source/Core/Common/Network.cpp

```diff
@@ -563,7 +572,7 @@ const char* DecodeNetworkError(s32 error_code)
                  sizeof(buffer), nullptr);
   return buffer;
 #elif defined(IS_BSD_STRERROR) ||                                                                  \
-    ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE)
+    ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE) || defined(__MUSL__)
```

Comment:
If it aims to use the XSI version, the check you added can be moved above and the define `IS_BSD_STRERROR` renamed to `USE_XSI_STRERROR_R` or similar.

Reply:
Fixed by renaming the macro to ``USE_XSI_STRERROR`` and by checking if the now-named ``__NO_GLIBC`` macro is defined.